src/python/bot/tasks/fuzz_task.py

```diff
@@ -1755,6 +1758,11 @@ def do_blackbox_fuzzing(self, fuzzer, fuzzer_directory, job_type):
         gestures = testcases_metadata[testcase_file_path]['gestures']
 
         env_copy = environment.copy()
+
+        # For some binaries, we specify trials, which are sets of flags that we
+        # only apply some of the time. Adjust APP_ARGS for them if needed.
+        trial_selector.setup_additional_args_for_app(env_copy)
```

Comment:
This seems wrong, i should have looked closely before. Trials are done once per fuzzing session and not every testcase, why do you do this?

Reply:
That's exactly what I'm trying to change here. See change description. I've also shared a doc with you with more discussion about this.